src/libstore/build/local-derivation-goal.cc

```diff
@@ -1464,6 +1464,11 @@ struct RestrictedStore : public virtual RestrictedStoreConfig, public virtual Lo
 
     virtual void addBuildLog(const StorePath & path, std::string_view log) override
     { unsupported("addBuildLog"); }
+
+    bool checkTrustedUser() override
+    {
+        throw Error("You cannot query trusted-user with recursive-nix");
```

Comment:
It may make sense to call recursive nix not trusted.

Reply:
In which case, should we just return false?

Reply:
I've taken the suggestion from @thufschmitt to include a default implementation of `unsupported("foo")` instead. If we should return false, that is something we can do in another PR, if this gets merged soon.

Reply:
> just return false?

Yes.

Reply:
I've returned `false` for the case of recursive nix now.